llvm/lib/Transforms/Scalar/CorrelatedValuePropagation.cpp

```diff
@@ -288,24 +288,36 @@ static bool processICmp(ICmpInst *Cmp, LazyValueInfo *LVI) {
   if (!Cmp->getOperand(0)->getType()->isIntOrIntVectorTy())
     return false;
 
-  if (!Cmp->isSigned())
+  if (!(Cmp->isSigned() || (Cmp->isUnsigned() && !Cmp->hasSameSign())))
```

Comment:
```suggestion
  if (!Cmp->isSigned() && (!Cmp->isUnsigned() || Cmp->hasSameSign())))
```
nit: Prefer pushing `!` into conditions.